org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/RefactorProcessor.java

```diff
@@ -153,12 +153,25 @@ public RefactorProcessor(PreferenceManager preferenceManager) {
 
 				getConvertForLoopProposal(context, coveringNode, proposals);
 				getAssignToVariableProposals(context, coveringNode, locations, proposals, params);
+				getIntroduceParameterProposals(params, context, coveringNode, locations, proposals);
 			}
 			return proposals;
 		}
 		return Collections.emptyList();
 	}
 
+	private boolean getIntroduceParameterProposals(CodeActionParams params, IInvocationContext context, ASTNode coveringNode, IProblemLocationCore[] locations, ArrayList<ChangeCorrectionProposal> resultingCollections) throws CoreException {
+		if (resultingCollections == null) {
+			return false;
+		}
+		CUCorrectionProposal proposal = RefactorProposalUtility.getIntroduceParameterRefactoringProposals(params, context, coveringNode, this.preferenceManager.getClientPreferences().isAdvancedIntroduceParameterRefactoringSupported(), locations);
+		if (proposal != null) {
+			resultingCollections.add(proposal);
```

Comment:
or `return resultingCollections.add(proposal);`?

Reply:
getIntroduceParameterProposals must return boolean.

Reply:
yes and 
```
boolean java.util.Collection.add(E e)

Ensures that this collection contains the specified element (optional operation). Returns true if this collection changed as a result of the call. 
```

Reply:
Fixed.